pkg/scheduler/core/scheduling_queue.go

```diff
@@ -484,29 +484,17 @@ var _ = UnschedulablePods(&UnschedulablePodsMap{})
 
 // Add adds a pod to the unschedulable pods.
 func (u *UnschedulablePodsMap) Add(pod *v1.Pod) {
-	podKey := u.keyFunc(pod)
-	if _, exists := u.pods[podKey]; !exists {
-		u.pods[podKey] = pod
-	}
+	u.pods[u.keyFunc(pod)] = pod
 }
 
 // Delete deletes a pod from the unschedulable pods.
 func (u *UnschedulablePodsMap) Delete(pod *v1.Pod) {
-	podKey := u.keyFunc(pod)
-	if _, exists := u.pods[podKey]; exists {
-		delete(u.pods, podKey)
-	}
+	delete(u.pods, u.keyFunc(pod))
 }
 
 // Update updates a pod in the unschedulable pods.
 func (u *UnschedulablePodsMap) Update(pod *v1.Pod) {
-	podKey := u.keyFunc(pod)
-	_, exists := u.pods[podKey]
-	if !exists {
-		u.Add(pod)
-		return
-	}
-	u.pods[podKey] = pod
+	u.Add(pod)
```

Comment:
seems like now both Add and Update are same, why to have both?

Reply:
Yes, they are the same now. The reason to have both is mostly readability. Having only one and calling "Add()" in places where an "update" is needed may make the code a bit confusing.

Reply:
That's fine, though I'd prefer AddOrUpdate and to avoid a slight chance of getting Add updated in future in such a way that might break Update unintentionally.

Reply:
Sure. Done. I also made the interface of the "UnschedulablePodsMap" private.